packages/hoppscotch-sh-admin/src/components/users/Details.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <div class="rounded-md">
+    <div class="grid gap-6 mt-4">
+      <div v-if="user.photoURL" class="relative h-20 w-20">
+        <img class="object-cover rounded-3xl mb-3" :src="user.photoURL" />
+        <span
+          v-if="user.isAdmin"
+          class="absolute left-16 bottom-0 text-xs font-medium px-3 py-0.5 rounded-full bg-green-900 text-green-300"
+        >
+          {{ t('users.admin') }}
+        </span>
+      </div>
+
+      <div v-else class="bg-primaryDark w-16 p-3 rounded-2xl mb-3 relative">
+        <icon-lucide-user class="text-4xl" />
+        <span
+          v-if="user.isAdmin"
+          class="absolute left-16 bottom-0 text-xs font-medium px-3 py-0.5 rounded-full bg-green-900 text-green-300"
+        >
+          {{ t('users.admin') }}
+        </span>
+      </div>
+
+      <div v-if="user.uid">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.uid')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ user.uid }}
+        </div>
+      </div>
+      <div>
+        <label class="text-secondaryDark" for="username">{{
+          t('users.name')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          <span v-if="user.displayName">
+            {{ user.displayName }}
+          </span>
+          <span v-else> {{ t('users.unnamed') }} </span>
+        </div>
+      </div>
+      <div v-if="user.email">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.email')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ user.email }}
+        </div>
+      </div>
+      <div v-if="user.createdOn">
+        <label class="text-secondaryDark" for="username">{{
+          t('users.created_on')
+        }}</label>
+        <div
+          class="w-full p-3 mt-2 bg-divider border-gray-600 rounded-md focus:border-emerald-600 focus:ring focus:ring-opacity-40 focus:ring-emerald-500"
+        >
+          {{ getCreatedDateAndTime(user.createdOn) }}
+        </div>
+      </div>
+    </div>
```

Comment:
This is an ideal candidate where the markup can be constructed based on state maintained locally. For instance, have an object like

```js
const userInfo = {
    uid: { 
       condition: user.uid,
       labelText: t('users.uid'),
       value: user.uid
     }
     ...
  }
```

Iterate over the objects and construct the markup. The `condition` field can be applied to the `v-if` directive.

Reply:
Thanks for the suggestion. Helped in reducing the markup significantly